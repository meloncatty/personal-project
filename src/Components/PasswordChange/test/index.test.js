import React from 'react'
import { PasswordChange } from '../'
import { shallow } from 'enzyme'

describe("PasswordChange", () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<PasswordChange />)
    expect(wrapper).toMatchSnapshot()
  });

  it("should have default state", () => {
    const wrapper = shallow(<PasswordChange />)
    const expected = {
      password: '',
      passwordAuth: '',
      submitSuccessful: false,
      error: false
    }

    expect(wrapper.state()).toEqual(expected)
  });

  it("should update state on submit", () => {
    const wrapper = shallow(<PasswordChange />)
    const mockEvent = {preventDefault: () => {}}
    const expected = {
      password: '',
      passwordAuth: '',
      submitSuccessful: false,
      error: false
    }
    wrapper.find('button').simulate('click', mockEvent)

    expect(wrapper.state()).toEqual(expected)
  });
});
