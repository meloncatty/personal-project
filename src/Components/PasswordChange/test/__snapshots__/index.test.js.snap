// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChange should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordChange />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="change-password"
        onSubmit={[Function]}
>
        <label>
                Change Password
        </label>
        <input
                className="password"
                onChange={[Function]}
                placeholder="New Password"
                type="password"
                value=""
        />
        <input
                className="password-confirm"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
        />
        <button
                disabled={true}
                type="submit"
        >
                Change Password
        </button>
        <p />
</form>,
      "className": "form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Change Password
</label>,
          <input
            className="password"
            onChange={[Function]}
            placeholder="New Password"
            type="password"
            value=""
/>,
          <input
            className="password-confirm"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Change Password
</button>,
          <p />,
          false,
        ],
        "className": "change-password",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Change Password",
          },
          "ref": null,
          "rendered": "Change Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password",
            "onChange": [Function],
            "placeholder": "New Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-confirm",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Change Password",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Change Password",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        false,
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="change-password"
          onSubmit={[Function]}
>
          <label>
                    Change Password
          </label>
          <input
                    className="password"
                    onChange={[Function]}
                    placeholder="New Password"
                    type="password"
                    value=""
          />
          <input
                    className="password-confirm"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Change Password
          </button>
          <p />
</form>,
        "className": "form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Change Password
</label>,
            <input
              className="password"
              onChange={[Function]}
              placeholder="New Password"
              type="password"
              value=""
/>,
            <input
              className="password-confirm"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Change Password
</button>,
            <p />,
            false,
          ],
          "className": "change-password",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Change Password",
            },
            "ref": null,
            "rendered": "Change Password",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password",
              "onChange": [Function],
              "placeholder": "New Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password-confirm",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Change Password",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Change Password",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          false,
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
